Feature view posts feature

ZenGelvs opened the pull request.

ZenGelvs committed:
Create viewposts.blade.php

ZenGelvs committed:
Added Views

ZenGelvs committed:
Added Views

ericnicdao commented on routes/web.php:
```diff
@@ -14,5 +16,15 @@
+
+//From Index return to create post
+Route::get('/createpost-from-viewpost', [PostController::Class, 'return']);
+
```
Please use Laravel route grouping for naming and prefixing routes.
https://laravel.com/docs/10.x/routing#route-groups
https://laravel.com/docs/10.x/routing#route-group-prefixes
https://laravel.com/docs/10.x/routing#route-group-name-prefixes

Also, as much as possible and if it fits, name your prefix and action/method as a resource.
https://laravel.com/docs/10.x/controllers#actions-handled-by-resource-controller

ericnicdao commented on routes/web.php:
```diff
@@ -14,5 +16,15 @@
 
 Route::get('/', function () {
-    return view('welcome');
+    return view('/addpost');
```
You can remove this for now.
Please see comment below on where to transfer adding posts.

ericnicdao commented on resources/views/welcome.blade.php:
```diff
@@ -4,137 +4,27 @@
+        </form>
+             </div> 
     </body>
 </html>
```
For all of your views. Please use layout templating.
https://laravel.com/docs/10.x/blade#layouts-using-template-inheritance

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,32 @@
+        ]);
+        
+        $incomingFields['title'] = strip_tags($incomingFields['title']);
+        $incomingFields['body'] = strip_tags($incomingFields['body']);
```
No need to use strip_tags.
As far as I know this is already handled by Laravel's Request class automatically.

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,32 @@
+        $incomingFields = $request->validate([
+            'body' => 'required',
+            'title' => 'required'
+        ]);
```
Please use FormRequest classes to handle your validations.
https://laravel.com/docs/10.x/validation#form-request-validation

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,32 @@
+    public function return(){
+        return view ('/addpost');
+    }
+}
```
Please use names from Laravel resource controller as much as possible.
This is related to the routes comment.
https://laravel.com/docs/10.x/controllers#actions-handled-by-resource-controller

ZenGelvs committed:
View Layouts

ZenGelvs committed:
Routes and Views

ericnicdao commented on routes/web.php:
```diff
@@ -14,5 +16,25 @@
+
+
+//Resource Controller
+Route::resource('posts', PostController::class);
```
Please remove. No need for this since you're already using/overriding the actions below manually.

ericnicdao commented on routes/web.php:
```diff
@@ -14,5 +16,25 @@
+Route::resource('posts', PostController::class);
+
+//Routegroup
+Route::group(['prefix'=> 'Posts.', 'as' => 'POSTS.'], function(){
```
Change 'Posts.' to 'posts' and 'POSTS.' to 'post.'

ericnicdao replied:
Prefix still has '.' value  on 'Posts.'.
Please remove dot and use lowercase only.

For "as" value.
Change to lowercase and remove 's' before the dot.
To 'post.'

ericnicdao commented on routes/web.php:
```diff
@@ -14,5 +16,25 @@
+//Routegroup
+Route::group(['prefix'=> 'Posts.', 'as' => 'POSTS.'], function(){
+
+    Route::post('/create-post', [PostController::class, 'store']) ->name('store');
```
Please change '/create-post' to just '/create'.
No need to add 'posts' since we already have it in the URL.
Please apply to others below with '-post'.

ericnicdao commented on resources/views/edit.blade.php:
```diff
@@ -0,0 +1,19 @@
+            <button class="mt-4 text-gray-500 dark:text-gray-400 c leading-relaxed">Save</button>
+            </form>
+
+            <form action="{{route('POSTS.return')}}" method='GET'> 
```
Use anchor tags instead of buttons.

ericnicdao commented on resources/views/posts.blade.php:
```diff
@@ -0,0 +1,18 @@
+        <button class="mt-4 text-gray-500 dark:text-gray-400 c leading-relaxed"> Post </button>
+        
+    </form>
+        <form action='{{route('POSTS.index')}}' method='POST'> 
```
Use anchor tags instead of buttons.

ericnicdao commented on app/Http/Requests/StorePostRequest.php:
```diff
@@ -0,0 +1,25 @@
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class StorePostRequest extends FormRequest
```
Please rename to StoreRequest.
We usually put this in subdirectories based on the user role then model.
So for example,
App\Http\Requests\User\Post\StoreRequest;
App\Http\Requests\Admin\Post\StoreRequest;

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    //edit function that retries data from db
+    public function edit(Post $posts){
```
Please rename to $post instead of $posts to fix plurality.

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    public function update(Post $posts, StorePostRequest $request){
+        $incomingFields = $request->validated();
```
Please use $data if you'll be assigning form data to a variable.

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    //return to addpost view
+    public function return(){
```
Please remove this whole method. This is not needed.

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,51 @@
+
+    public function destroy(Post $posts){
+        $posts->delete();
+        return redirect('/');
```
Please use route name to redirect.

ericnicdao replied:
Please use redirect()->back() or redirect()->route('route.name');

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,51 @@
+        
+        $incomingFields = $request->validated();
+        Post::create($incomingFields);
+       return view ('posts');
```
Please return redirect instead of returning a view.

ericnicdao commented on app/Http/Requests/StorePostRequest.php:
```diff
@@ -0,0 +1,25 @@
+    public function rules(): array
+    {
+        return [
+                'body' => 'required',
```
Please add other appropriate rules for the field.

ericnicdao commented on resources/views/layout/master.blade.php:
```diff
@@ -0,0 +1,30 @@
+        </div>
+
+        <div class="container">
+            @yield('Editpost')
```
Please only use one yield directive.
This can be reused.
We usually name it 'content'
@yield('content')

Then on your child blades just call it as
@section('content')
@endsection

ericnicdao commented on resources/views/posts.blade.php:
```diff
@@ -0,0 +1,18 @@
+        <button class="mt-4 text-gray-500 dark:text-gray-400 c leading-relaxed"> View Posts </button>
+        </form>
+</div> 
+@stop
```
Please use @endsection.

ericnicdao commented on resources/views/view.blade.php:
```diff
@@ -0,0 +1,44 @@
+
+    </div>  
+</div> 
+@stop
```
Please use @endsection.

ericnicdao commented on resources/views/view.blade.php:
```diff
@@ -0,0 +1,44 @@
+        <div  style= "padding:10px; margin:10px" > 
+            <h1 class="mt-6 text-xl font-semibold text-gray-900 dark:text-white" > All posts  </h1>
+
+            @foreach($posts as $posts)
```
If you can please transfer the lists to where your create form is currently located.
Make it look like Facebook or Reddit or any other social media platform.

ericnicdao commented on resources/views/posts.blade.php:
```diff
@@ -0,0 +1,18 @@
+@extends('layout.master')  
```
Please rename this file to index.blade.php

ZenGelvs committed:
Fixed variable names

ZenGelvs committed:
Update StoreRequest.php

ZenGelvs committed:
Create and view post joined

ZenGelvs committed:
Removed return method

ZenGelvs committed:
Anchors

Post and delete are still buttons
Edit>Save button not working
Common errors get methods.

ZenGelvs committed:
Restore

ZenGelvs committed:
Update edit.blade.php

ZenGelvs committed:
Fixed update feature with renaming $post

ZenGelvs committed:
Fixed put, store, and delete

ZenGelvs committed:
Update view.blade.php

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,44 @@
+        
+        $data = $request->validated();
+        Post::create($data);
+        return redirect('/Posts./view');
```
Please use redirect()->back() or redirect()->route('route.name');

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,44 @@
+
+        $post->update($data);
+
+        return view ('posts');
```
Please use redirect()->back() or redirect()->route('route.name');

ericnicdao commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,44 @@
+
+    public function destroy(Post $post){
+        $post->delete();
+        return redirect('/Posts./view');
```
Please use redirect()->back() or redirect()->route('route.name');

ericnicdao commented on resources/views/edit.blade.php:
```diff
@@ -0,0 +1,18 @@
+@extends('layout.index')  
+@section('Content')
+<div> 
+    
```
Please fix spacings and indentations for all views.

ZenGelvs committed:
Proper Redirects

ZenGelvs committed:
Update edit.blade.php

ZenGelvs committed:
Ready for Merge

ZenGelvs committed:
Merge branch 'master' of https://github.com/ZenGelvs/Gelvins_Blog into feature-view-posts-feature

ZenGelvs committed:
Revert "Merge branch 'master' of https://github.com/ZenGelvs/Gelvins_Blog into feature-view-posts-feature"

This reverts commit d77a21cf38a73cc9a36f0d2a6137b8721fa72d9f, reversing
changes made to 60455417a185b1adb6057aec816e1420b5c9eed4.

ZenGelvs committed:
Update composer.lock

ZenGelvs committed:
Revert "Update composer.lock"

This reverts commit 341a973316ed21932455b244f44196731a001d0b.

ZenGelvs committed:
{post} route group added

ZenGelvs committed:
API

ZenGelvs committed:
API and Resources

ZenGelvs committed:
API post controller

ZenGelvs committed:
API json and Resources

ZenGelvs committed:
Update Handler.php

ZenGelvs committed:
Implicit Binding, spacing, and indentation

ZenGelvs committed:
API

ZenGelvs committed:
Update api.php

ZenGelvs committed:
Update web.php

ZenGelvs committed:
Merge pull request #5 from ZenGelvs/feature-API

Feature api

YutaInouePH commented on app/Http/Controllers/Api/PostController.php:
```diff
@@ -0,0 +1,50 @@
+{
+    //Retrieve all Posts
+    public function index(){
+        return PostResource::collection(Post::all());
```
Make sure to paginate

YutaInouePH commented on app/Http/Controllers/Api/PostController.php:
```diff
@@ -0,0 +1,50 @@
+
+class PostController extends Controller
+{
+    //Retrieve all Posts
```
Have proper PHP doc with param and return types

YutaInouePH commented on app/Http/Resources/PostResource.php:
```diff
@@ -0,0 +1,23 @@
+    {
+        return [
+            'id' => $this->id,
+            'Title' => $this->Title,
+            'Body' => $this->Body
```
usually, the coding standard is to use small alphabet for the first character

YutaInouePH commented on database/migrations/2023_07_25_084752_create_posts_table.php:
```diff
@@ -0,0 +1,29 @@
+    {
+        Schema::create('posts', function (Blueprint $table) {
+            $table->id();
+            $table->string('Title');
+            $table->longText('Body');
```
column names should be camel case

ZenGelvs committed:
Pagination and code standard

ZenGelvs committed:
Added param and return types

ZenGelvs committed:
PhpDoc and Pint

ZenGelvs committed:
PHPDoc and Pint

ZenGelvs committed:
Php Doc

ZenGelvs committed:
PHP Doc
